### PR TITLE
Socket implementation and beginning of Quiz Implementation

### PR DESCRIPTION
**Pull request by sammyjh30**
PLEASE NOTE THAT THIS IS NOT COMPLETE. A lot has broken with the quiz viewing setup. This is to get the sockets into develop and list the quizes

### CODE REVIEW

File: QuizMakerAngular/src/app/components/chat-inbox/chat-inbox.component.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit, Input } from '@angular/core';
+import * as io from 'socket.io-client'
+import { AuthService } from "../../services/auth.service";
+
+const SOCKET_ENDPOINT = 'localhost:3001';
```
Comment (KevaKevz) [inline QuizMakerAngular/src/app/components/chat-inbox/chat-inbox.component.ts:5]: We need to look at where this is when hosted and probably move to the config files

File: QuizMakerAngular/src/app/components/dashboard/dashboard.component.html [hunk 2]
```diff
@@ -14,4 +14,11 @@ <h1>User Profile</h1>
   <h1>Hello: <strong>{{(user.displayName) ? user.displayName : 'User'}}</strong></h1>
   <p>User ID: <strong>{{user.uid}}</strong></p>
   <p>Email: <strong>{{user.email}}</strong></p>
+  <p>Email Verified: <strong>{{authService.token}}</strong></p>
+
+  <!-- For each quiz in the user's quiz list -->
+  <!-- <a [routerLink]="['/quiz-session', ‘quiz.id’]">{{quiz.name}} </a> -->
+  <a routerLink="/quiz-session/1/123">First Quiz</a>
+  <a routerLink="/quiz-session/1">First Quiz</a>
+  <a routerLink="/quiz-session/2">Second Quiz</a>
```
Comment (callummdunne) [inline QuizMakerAngular/src/app/components/dashboard/dashboard.component.html:23]: Will need to change in next PR but thats chilled

**Review by callummdunne — APPROVED** [review]

File: QuizMakerAngular/src/app/components/quiz-session/quiz-session.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { Component, OnInit, Input } from '@angular/core';
+import {ActivatedRoute} from "@angular/router";
+import { AuthService } from "../../services/auth.service";
+import * as io from 'socket.io-client'
+// import { User } from 'firebase';
+import { User } from '../../models/user';
+
+const SOCKET_ENDPOINT = 'localhost:3001';
```
Comment (KevaKevz) [inline QuizMakerAngular/src/app/components/quiz-session/quiz-session.component.ts:8]: Yip

File: QuizMakerAngular/src/app/components/dashboard/dashboard.component.html (hunk 2 above)
Comment (sammyjh30) [reply QuizMakerAngular/src/app/components/dashboard/dashboard.component.html:23]: This page in theory will never be seen, so yeah

File: QuizMakerAngular/src/app/services/user.service.ts [hunk 4]
```diff
@@ -136,9 +136,9 @@ export class UserService {
     return this.http.post(this.link + '/removeTeamMember', data, this.httpOptions).toPromise();
   }
 
-  getUserDashboard(userId: number): Promise<User[]> {
-    let url = `${this.link}/getTeamMembers/${userId}`;
-    return this.http.get<User[]>(url, this.httpOptions).toPromise();
+  getUserDashboard(userId: string): Promise<User[]> {
+    let url = `${this.link}/dashboard/${userId}`;
+    return this.http.get<any>(url, this.httpOptions).toPromise();
   }
```
Comment (KevaKevz) [inline QuizMakerAngular/src/app/services/user.service.ts:142]: Why this change?

File: QuizMakerAngular/src/app/components/chat-inbox/chat-inbox.component.ts (hunk 1 above)
Comment (sammyjh30) [reply QuizMakerAngular/src/app/components/chat-inbox/chat-inbox.component.ts:5]: Yeah, that was the plan, but rather you deal with that!

File: QuizMakerAngular/src/app/services/user.service.ts (hunk 4 above)
Comment (callummdunne) [reply QuizMakerAngular/src/app/services/user.service.ts:142]: It doesnt actually return just user stuff. it resturns a bunch of stuff for the quiz

File: QuizMakerAngular/src/chat/chat.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import * as io from 'socket.io-client';
+import { Observable } from 'rxjs/Observable';
+
+export class ChatService {
+    private url = 'http://localhost:3000';
+    private socket;
```
Comment (KevaKevz) [inline QuizMakerAngular/src/chat/chat.service.ts:6]: change to config

File: QuizMakerAngular/src/app/services/user.service.ts (hunk 4 above)
Comment (sammyjh30) [reply QuizMakerAngular/src/app/services/user.service.ts:142]: This is how we get the user's quiz list. Was it being used elsewhere?

File: QuizMakerNode/app.js [hunk 6]
```diff
@@ -19,6 +19,49 @@ app.get("/", (req, res) => {
...
+    // the rooms array contains at least the socket ID
+  });
+
+  socket.on('disconnect', () => {
+     console.log('a user disconnected. ' +  username);
+  });
+});
+
+
+http.listen(3001, () => {
```
Comment (KevaKevz) [inline QuizMakerNode/app.js:60]: We need to change this to port.env.host

**Review by KevaKevz — APPROVED** [review]
